extensions-contrib/prometheus-emitter/src/main/java/org/apache/druid/emitter/prometheus/PrometheusEmitterConfig.java

```diff
@@ -94,6 +101,11 @@ public PrometheusEmitterConfig(
     this.flushPeriod = flushPeriod;
     this.addHostAsLabel = addHostAsLabel;
     this.addServiceAsLabel = addServiceAsLabel;
+    this.extraLabels = extraLabels != null ? extraLabels : Collections.emptyMap();
+    // Validate label names early to prevent Prometheus exceptions later.
+    for (String key : this.extraLabels.keySet()) {
+      Preconditions.checkArgument(PATTERN.matcher(key).matches(), "Invalid metric label name: " + key);
```

Comment:
Perhaps we can use the `DruidException` class to surface this error to the operator: 

```suggestion
      throw DruidException.forPersona(DruidException.Persona.OPERATOR)
                          .ofCategory(DruidException.Category.INVALID_INPUT)
                          .build(
                              StringUtils.format(
                                  "Invalid metric label name [%s]. Label names must conform to the pattern [%s].",
                                  key,
                                  PATTERN.pattern()
                              )
                          );
```